docarray/documents/helper.py

```diff
@@ -0,0 +1,120 @@
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple, Type, Union
+
+from pydantic import create_model, create_model_from_typeddict
+from pydantic.config import BaseConfig
+from typing_extensions import TypedDict
+
+from docarray import BaseDocument
+
+if TYPE_CHECKING:
+    from pydantic.typing import AnyClassMethod
+
+
+def create_doc(
+    __model_name: str,
+    *,
+    __config__: Optional[Type[BaseConfig]] = None,
+    __base__: Union[
+        None, Type['BaseDocument'], Tuple[Type['BaseDocument'], ...]
+    ] = BaseDocument,
+    __module__: str = __name__,
+    __validators__: Dict[str, 'AnyClassMethod'] = None,  # type: ignore
+    __cls_kwargs__: Dict[str, Any] = None,  # type: ignore
+    __slots__: Optional[Tuple[str, ...]] = None,
+    **field_definitions: Any,
+):
```

Comment:
would be nice to add the return type.

I suggested introducing a TypeVar that you can use in `__base__` and use in the return statement

Reply:
okay I removed the return type because of mypy error.
I'll try to find a way for both adding the return and pass mypy.